linode/manager
# Upcoming: [DI-20844] - Tooltip for widget level filters and icons, beta feedbacks and CSS changes for CloudPulse

## Description 📝
We have added Tooltip for all of our icons and widget level filters. Also addressed some of the beta demo feedbacks like
CamelCase for Aggregation Function dropdown label, Zeroth state of time duration and CSS updates for legend rows inside the widget.

## Changes  🔄

1. Added Tooltip for all icons and widget level filters
2. CamelCase labels for Aggregation Function dropdown in widget
3. Zero state update for time duration to move to Last 30 Minutes
4. Padding and minHeight adjustment for legend rows inside widget
5. Widget Unit scale up display logic update
6. CloudpulseResourceSelection, publish the selected values only if the autocomplete closes since it is a multiselect component and reduce metrics call fired.
7. Interval selection should default to AUTO if there are no preferences in its initial state

## Target release date 🗓️
10-10-2024

## Preview 📷
| Before  | After   |
| ------- | ------- |
|![Screenshot 2024-10-08 at 3 54 12 PM](https://github.com/user-attachments/assets/98f43a78-f0bd-4e0d-8bd6-609accca20e7)|![Screenshot 2024-10-08 at 3 54 31 PM](https://github.com/user-attachments/assets/58879eaf-9eb1-44ed-a8a0-7024f4a728de)|
|![Screenshot 2024-10-08 at 3 54 34 PM](https://github.com/user-attachments/assets/7a60e87c-9d70-422e-a0a4-0f2e1d28a0cf)|![Screenshot 2024-10-08 at 3 54 36 PM](https://github.com/user-attachments/assets/d9b9d337-e988-4ffa-a4a7-384cd6e34549)|
|![Screenshot 2024-10-08 at 3 54 48 PM](https://github.com/user-attachments/assets/8d839c05-2dc5-4d9c-bf23-a5d4d183382c)|![Screenshot 2024-10-08 at 3 54 51 PM](https://github.com/user-attachments/assets/43e5275f-d395-4e4e-88b8-485dc6058c87)|
|![Screenshot 2024-10-08 at 3 55 04 PM](https://github.com/user-attachments/assets/530d4897-a304-433c-bf83-3149a0a1f6fc)|![Screenshot 2024-10-08 at 3 55 22 PM](https://github.com/user-attachments/assets/4a1afa14-a271-426f-9ef7-28cb8927df40)|
|![Screenshot 2024-10-08 at 7 00 20 PM](https://github.com/user-attachments/assets/98984b1a-e077-494b-bad8-3be838a89de9)|![Screenshot 2024-10-08 at 4 40 51 PM](https://github.com/user-attachments/assets/03e43f9d-44ce-4ba8-9fcb-54ac3c8be134)|
|![Screenshot 2024-10-08 at 6 58 49 PM](https://github.com/user-attachments/assets/b1e45eee-f645-46c2-ae67-83cf27f52920)|![Screenshot 2024-10-08 at 3 26 54 PM](https://github.com/user-attachments/assets/1f797e86-1997-4cc0-96b3-e8a20b2e9ad8)|


## How to test 🧪

1. Login as mock user as some endpoints are not ready
2. Navigate to Monitor page
3. Select dashboard and necessary filters to view the widgets
4. You can see tooltips for icons and widget level filters
5. Also you can see aggregate function dropdown displaying labels in camelcase
9. Once you clear the selected dashboard time duration returns to Last 30 Minutes

## As an Author I have considered 🤔

*Check all that apply*
